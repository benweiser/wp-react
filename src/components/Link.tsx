import * as React from "react";

export interface Props {
    href: string;
    linkText: string;
    target?: string;
}

const Link: React.SFC<Props> = ({ href, linkText, target = "_self" }: Props): JSX.Element => {
    return (
        <a href={href} target={target}>{linkText}</a>
    );
};

export default Link;