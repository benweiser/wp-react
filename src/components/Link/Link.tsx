import * as React from "react";

export interface LinkProps {
  readonly href: string;
  readonly linkText: string;
  readonly target?: string;
}

const Link: React.SFC<LinkProps> = ({
  href,
  linkText,
  target = "_self"
}: LinkProps): JSX.Element => {
  return (
    <a href={href} target={target}>
      {linkText}
    </a>
  );
};

export default Link;
