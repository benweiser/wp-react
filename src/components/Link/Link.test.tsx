import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Link from "./Link";

it("Link component renders correctly", () => {
  const wrapper = shallow(
    <Link href={"http://www.google.com"} linkText={"Google"} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
