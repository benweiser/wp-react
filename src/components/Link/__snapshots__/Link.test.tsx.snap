// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly 1`] = `
<a
  href="http://www.google.com"
  target="_self"
>
  Google
</a>
`;
